src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,34 @@
+import { popupSaveButton } from "../utils/constants";
+import Popup from "./Popup";
+
+export default class PopupWithConfirmation extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupSaveButton = this._popupElement.querySelector(
+      ".popup__save-button"
+    );
+    this._popupSaveButtonText = this._popupSaveButton.textContent;
+  }
+
+  close() {
+    this._popupSaveButton.removeEventListener("click", this._handleFormSubmit);
+    super.close();
+  }
+
+  setSubmitAction(fn) {
```

Comment:
Great job implementing confirmation popup and the functionality related to it! In general, it can support any confirmation.